app/search_builders/sufia/find_works_search_builder.rb

```diff
@@ -0,0 +1,44 @@
+class Sufia::FindWorksSearchBuilder < Sufia::SearchBuilder
+  include Sufia::MySearchBuilderBehavior
+  include CurationConcerns::FilterByType
+
+  def initialize(context)
+    super(context)
+    @id = context.params[:id]
+    @q = context.params[:q]
+  end
+
+  self.default_processor_chain += [:filter_on_title, :show_only_resources_deposited_by_current_user]
+  self.default_processor_chain += [:show_only_other_works, :show_only_works_not_child, :show_only_works_not_parent]
+
+  def filter_on_title(solr_parameters)
+    solr_parameters[:fq] ||= []
+    solr_parameters[:fq] += [ActiveFedora::SolrQueryBuilder.construct_query(title_tesim: @q)]
+  end
+
+  def show_only_other_works(solr_parameters)
+    solr_parameters[:fq] ||= []
+    solr_parameters[:fq] += [
+      "-" + ActiveFedora::SolrQueryBuilder.construct_query_for_ids([@id])
+    ]
+  end
+
+  def show_only_works_not_child(solr_parameters)
+    ids = ActiveFedora::SolrService.query("{!field f=id}#{@id}", fl: "member_ids_ssim").flat_map { |x| x.fetch("member_ids_ssim", []) }
```

Comment:
make sure you pass `:rows` to `ActiveFedora::SolrService.query` or it'll only return the first 10.